tests/cucumber/e2e/stepDefinations/loginContext.js

```diff
@@ -0,0 +1,32 @@
+const { Given, When, Then } = require("@cucumber/cucumber");
+const { LoginPage } = require("../pageObjects/LoginPage.js");
+const assert = require("assert");
+const loginPage = new LoginPage();
+
+Given("user has navigated to the login page", async function () {
+    await loginPage.goToLoginPage();
+    assert.equal(page.url(), loginPage.url, `not in login page`);
```

Comment:
You can just check using expect() to see whether user navigates to user page or not.
```suggestion
    expect(page.url()).toBe(loginPage.url)
```